OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/DbxFileActions.java

```diff
@@ -0,0 +1,115 @@
+package latexstudio.editor;
+
+import com.dropbox.core.DbxClient;
+import com.dropbox.core.DbxEntry;
+import com.dropbox.core.DbxException;
+import com.dropbox.core.DbxWriteMode;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.swing.DefaultListSelectionModel;
+import javax.swing.JList;
+import javax.swing.JOptionPane;
+import latexstudio.editor.files.FileService;
+import latexstudio.editor.remote.DbxEntryDto;
+import latexstudio.editor.remote.DbxState;
+import latexstudio.editor.remote.DbxUtil;
+import latexstudio.editor.util.ApplicationUtils;
+import org.apache.commons.io.IOUtils;
+import org.openide.util.Exceptions;
+
+public class DbxFileActions {
+
+    private EditorTopComponent etc;
+
+    public DbxFileActions(EditorTopComponent etc) {
+        this.etc = etc;
+    }
+
+    public void saveProgress(DropboxRevisionsTopComponent drtc) {
+        DbxClient client = DbxUtil.getDbxClient();
```

Comment:
It would make sense to put `DbxClient client` as an additional Parameter, because it would be easier to write a test with a moked DbxClient. 
Refactor with 'Introduce Parameter'